### PR TITLE
Implemented category adding functionality

### PR DESCRIPTION
**Pull request by TSlashDreamy**
## Implemented (11.10.23) 1fbbd26c0cca0a60799c793ac14929e2e299d6c2
<details>
<summary>Preview</summary>

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/45912519/e1892d5a-e997-405d-984a-aadb76809e25

</details>

<details>
<summary>Lint</summary>

зображення

</details>

### CODE REVIEW

**Comment by abalanovsky** [comment]
Can u increase coverage?

File: src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const { data, errors, handleInputChange, handleBlur, handleSubmit } =
+    useForm<CreateCategoriesParams>({
+      initialValues: initialValues,
+      validations,
+      onSubmit: async () => {
+        setLoading(true)
+        await createCategories({
+          name: data.name
```
Comment (Tolik170) [inline src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx:41]: you can simpy pass data to createCategories:
createCategories(data)

File: src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      </Typography>
+
+      <Box sx={styles.form}>
+        <AppTextField
+          errorMsg={t(errors.name)}
+          fullWidth
+          label={t('myResourcesPage.categories.name')}
+          onBlur={handleBlur('name')}
+          onChange={handleInputChange('name')}
+          required
```
Comment (Tolik170) [inline src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx:65]: I guess you should pass value prop too

File: src/containers/my-resources/categories-container/CategoriesContainer.tsx [hunk 3]
```diff
@@ -89,6 +131,18 @@ const CategoriesContainer = () => {
     fetchOnMount: false
   })
 
+  const onAdd = () => {
+    openModal({
+      component: (
+        <AddCategoriesModal
+          closeModal={closeModal}
+          createCategories={(categoryName?: CreateCategoriesParams) =>
```
Comment (Tolik170) [inline src/containers/my-resources/categories-container/CategoriesContainer.tsx:139]: Can you just pass handleCreateCategory?
createCategories(handleCreateCategory)

and then, inside AddCategoriesModal pass category name to this callback

File: src/types/categories/interfaces/categories.interfaces.ts [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
 
 export interface CategoriesParams extends RequestParams {
   name: string
 }
+
+export interface CreateCategoriesParams {
+  name: string
+}
+
+export interface CreatedCategory extends CommonEntityFields {
```
Comment (Tolik170) [inline src/types/categories/interfaces/categories.interfaces.ts:12]: you can use "Categories" interface, it looks the same

File: src/types/categories/interfaces/categories.interfaces.ts [hunk 5]
```diff
@@ -1,5 +1,15 @@
+import { CommonEntityFields } from '~/types/common/common.index'
 import { RequestParams } from '~/types/services/types/services.types'
 
 export interface CategoriesParams extends RequestParams {
   name: string
 }
+
+export interface CreateCategoriesParams {
+  name: string
```
Comment (Tolik170) [inline src/types/categories/interfaces/categories.interfaces.ts:9]: maybe better to pick it from Categories or you can do like this: 
name: Categories['name']

**Review by maarekua — APPROVED** [review]

File: src/components/question-editor/QuestionEditor.tsx [hunk 6]
```diff
@@ -182,6 +182,32 @@ const QuestionEditor: FC<QuestionEditorProps> = ({ fetchData, loading }) => {
...
+            sx={styles.addButton}
+            variant={ButtonVariantEnum.Text}
+          >
+            <AddIcon />
+            {t('myResourcesPage.categories.addBtn')}
+          </AppButton>
+          <Divider sx={styles.divider} />
+        </Box>
+      )}
+      <Box component='li' {...props}>
```
[the remark this answers is not in the document]
Comment (TSlashDreamy) [reply src/components/question-editor/QuestionEditor.tsx]: > move li to ComponentEnum

moved and used it

File: src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx (hunk 1 above)
Comment (TSlashDreamy) [reply src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx]: > you can simpy pass data to createCategories: createCategories(data)

oh, you are right, that's better

File: src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx (hunk 2 above)
Comment (TSlashDreamy) [reply src/containers/my-resources/add-categories-modal/AddCategoriesModal.tsx:63]: > I guess you should pass value prop too

added value prop

File: src/containers/my-resources/categories-container/CategoriesContainer.tsx (hunk 3 above)
Comment (TSlashDreamy) [reply src/containers/my-resources/categories-container/CategoriesContainer.tsx]: > Can you just pass handleCreateCategory? createCategories(handleCreateCategory)
> 
> and then, inside AddCategoriesModal pass category name to this callback

Whoops, i forgot to remove this from here. It was already in AddCategoriesModal. Thanks 😅

File: src/types/categories/interfaces/categories.interfaces.ts (hunk 4 above)
Comment (TSlashDreamy) [reply src/types/categories/interfaces/categories.interfaces.ts]: > you can use "Categories" interface, it looks the same

You are right! Thanks for the advice

File: src/types/categories/interfaces/categories.interfaces.ts (hunk 5 above)
Comment (TSlashDreamy) [reply src/types/categories/interfaces/categories.interfaces.ts]: > maybe better to pick it from Categories or you can do like this: name: Categories['name']

Yup, that's better. Fixed

File: src/containers/my-resources/categories-container/CategoriesContainer.tsx [hunk 7]
```diff
@@ -53,6 +59,20 @@ const CategoriesContainer = () => {
     [setAlert]
   )
 
+  const onResponse = useCallback(
+    (response: Categories | null) => {
+      const categoryName = response ? response.name : ''
+
+      setAlert({
+        severity: snackbarVariants.success,
+        message: t('myResourcesPage.categories.successCreation', {
```
Comment (Tolik170) [inline src/containers/my-resources/categories-container/CategoriesContainer.tsx:68]: inside snackbar-context it passes this message param to translation function: t(message), here you pass this message already with translation, so will it work?

Comment (TSlashDreamy) [reply src/containers/my-resources/categories-container/CategoriesContainer.tsx:68]: > inside snackbar-context it passes this message param to translation function: t(message), here you pass this message already with translation, so will it work?

yeah, it's already working, you can see it in preview

**Review by Tolik170 — APPROVED** [review]

**Review by Mav-Ivan — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Review by OlyaKorchan — APPROVED** [review]

**Pull request merged**